beacon_node/eth2_libp2p/src/types/subnet.rs

```diff
@@ -21,8 +21,52 @@ impl PartialEq for SubnetDiscovery {
                             < DURATION_DIFFERENCE
                 }
                 (None, None) => true,
-                (None, Some(_)) => true,
-                (Some(_), None) => true,
+                (None, Some(_)) => false, // not equal
```

Comment:
Can you confirm you're not breaking attestation subnet logic here. 

This was implemented specifically for a reason. I recall this logic being somewhat critical in deciding if we need to search for new subnets or not. 

I agree the current form is not strictly correct from a `PartialEq` standpoint but perhaps its correct for its use in the attestation service

Reply:
> This was implemented specifically for a reason. I recall this logic being somewhat critical in deciding if we need to search for new subnets or not.

Thanks for pointing it out. I think this was needed before the attestation service refactor. Removing the `PartialEq` impl entirely only affected the tests att service tests. 

I'll run this on medalla for a bit and double check the logic to make sure.

Reply:
Okay, I'm convinced that we aren't using the `PartialEq` impl anywhere else but in the attestation service tests. 
For e.g. https://github.com/sigp/lighthouse/blob/master/beacon_node/network/src/attestation_service/tests/mod.rs#L208

Also ran this on medalla for about 2 hours with 0 missed attestations.
So I'm pretty sure this doesn't break any existing logic.

However, since we don't compare any 2 `SubnetDiscovery` structs directly, maybe we can remove the `PartialEq` impl entirely or have it behind a `#cfg(test)`  flag?

Reply:
Yeah, lets gate it behind the #cfg(test). 

Thanks :)